### PR TITLE
Format Console Output

### PR DESCRIPTION
**Pull request by dennis-behm**
This PR introduces a flag to configure a more human readable console output about the dependency resolution process of zAppBuild. 

Having `formatConsoleOutput` turned on in `build-conf/build.properties` or `application-conf/application.properties` as an individual setting, it will print in `--verbose` logging mode table views, rather than raw JSON .  

```
*** Building file MortgageApplication/cobol/epsnbrvl.cbl
*** Creating dependency resolver for MortgageApplication/cobol/epsnbrvl.cbl with [{"library": "SYSLIB", "searchPath": [ {"sourceDir": "/u/dbehm/userBuild", "directory": "MortgageApplication/copybook"} ]                }] rules
*** Scanning file with the default scanner
*** Resolution rules for MortgageApplication/cobol/epsnbrvl.cbl:
    Library   Category    SourceDir/File              Directory                           Collection              Archive
    --------- ----------- --------------------------- ----------------------------------- ----------------------- -------------------
    SYSLIB    N/A         /u/dbehm/userBuild          MortgageApplication/copybookN/A                     N/A
*** Physical dependencies for MortgageApplication/cobol/epsnbrvl.cbl:
    Library   Category        Name      Status        SourceDir/File
    --------- --------------- --------- ------------- -----------------------------------
    SYSLIB    COPY            EPSNBRPM  RESOLVED      MortgageApplication/copybook/epsnbrpm.cpy
   *SYSLIB    COPY            XYZ       NOT RESOLVED  N/A
Cobol compiler parms for MortgageApplication/cobol/epsnbrvl.cbl = LIB,ADATA,EX(ADX(ELAXMGUX))
```

### CODE REVIEW

**Comment by dennis-behm** [comment]
It is extended to support also the Impact Resolver configuration.

```*** Perform impacted analysis for changed files.
** Found build script mapping for MortgageApplication/cobol/epsmlist.cbl. Adding to build list
** Performing impact analysis on changed file MortgageApplication/cobol/epsmlist.cbl
*** Creating impact resolver for MortgageApplication/cobol/epsmlist.cbl with [{"library": "SYSLIB", "searchPath": [ {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/copybook"} ]                },{"library": "SYSLIB", "searchPath": [ {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/bms"} ]             },{"category": "LINK", "searchPath": [ {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/cobol"}, {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/link"} ]             },{"category": "PROPERTY"}] rules
    Collection
    -------------------
    MortgageApplication-dependencyResolverOutput
    MortgageApplication-dependencyResolverOutput-outputs
*** Configured resulution rules:
    Library   Category    SourceDir/File                                    Directory                           Collection              Archive
    --------- ----------- ------------------------------------------------- ----------------------------------- ----------------------- -------------------
    SYSLIB    N/A         /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/copybook        N/A                     N/A
    SYSLIB    N/A         /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/bms             N/A                     N/A
    N/A       LINK        /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/cobol           N/A                     N/A
    N/A       LINK        /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/link            N/A                     N/A
** Found impacted file MortgageApplication/link/epsmlist.lnk
** MortgageApplication/link/epsmlist.lnk is impacted by changed file MortgageApplication/cobol/epsmlist.cbl. Adding to build list.
** Performing impact analysis on changed file MortgageApplication/copybook/epsmtcom.cpy
*** Creating impact resolver for MortgageApplication/copybook/epsmtcom.cpy with [{"library": "SYSLIB", "searchPath": [ {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/copybook"} ]                },{"library": "SYSLIB", "searchPath": [ {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/bms"} ]             },{"category": "LINK", "searchPath": [ {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/cobol"}, {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/link"} ]             },{"category": "PROPERTY"}] rules
    Collection
    -------------------
    MortgageApplication-dependencyResolverOutput
    MortgageApplication-dependencyResolverOutput-outputs
*** Configured resulution rules:
    Library   Category    SourceDir/File                                    Directory                           Collection              Archive
    --------- ----------- ------------------------------------------------- ----------------------------------- ----------------------- -------------------
    SYSLIB    N/A         /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/copybook        N/A                     N/A
    SYSLIB    N/A         /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/bms             N/A                     N/A
    N/A       LINK        /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/cobol           N/A                     N/A
    N/A       LINK        /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/link            N/A                     N/A
** Found impacted file MortgageApplication/cobol/epscmort.cbl
** MortgageApplication/cobol/epscmort.cbl is impacted by changed file MortgageApplication/copybook/epsmtcom.cpy. Adding to build list.
** Found impacted file MortgageApplication/link/epsmlist.lnk
** MortgageApplication/link/epsmlist.lnk is impacted by changed file MortgageApplication/copybook/epsmtcom.cpy. Adding to build list.
** Found impacted file MortgageApplication/cobol/epsmlist.cbl
** MortgageApplication/cobol/epsmlist.cbl is impacted by changed file MortgageApplication/copybook/epsmtcom.cpy. Adding to build list.
** Found impacted file MortgageApplication/cobol/epscsmrt.cbl
** MortgageApplication/cobol/epscsmrt.cbl is impacted by changed file MortgageApplication/copybook/epsmtcom.cpy. Adding to build list.
** Calculation of impacted files by changed properties has been skipped due to configuration.
** Writing build list file to /u/dbehm/test-zapp-app-out/mortgageout/build.20220208.091414.014/buildList.txt
MortgageApplication/cobol/epsmlist.cbl
MortgageApplication/cobol/epscsmrt.cbl
MortgageApplication/cobol/epscmort.cbl
MortgageApplication/link/epsmlist.lnk
** Invoking build scripts according to build order: BMS.groovy,Cobol.groovy,LinkEdit.groovy
** Building files mapped to Cobol.groovy script
required props = cobol_srcPDS,cobol_cpyPDS,cobol_objPDS,cobol_loadPDS,cobol_compiler,cobol_linkEditor,cobol_tempOptions,applicationOutputsCollectionName,  SDFHCOB,SDFHLOAD,SDSNLOAD,SCEELKED
** Creating / verifying build dataset DBEHM.DBB.BUILD.COBOL
** Creating / verifying build dataset DBEHM.DBB.BUILD.COPY
** Creating / verifying build dataset DBEHM.DBB.BUILD.OBJ
** Creating / verifying build dataset DBEHM.DBB.BUILD.DBRM
** Creating / verifying build dataset DBEHM.DBB.BUILD.LOAD
*** Building file MortgageApplication/cobol/epsmlist.cbl
*** Creating dependency resolver for MortgageApplication/cobol/epsmlist.cbl with [{"library": "SYSLIB", "searchPath": [ {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/copybook"} ]                }] rules
*** Scanning file with the default scanner
*** Resolution rules for MortgageApplication/cobol/epsmlist.cbl:
*** Configured resulution rules:
    Library   Category    SourceDir/File                                    Directory                           Collection              Archive
    --------- ----------- ------------------------------------------------- ----------------------------------- ----------------------- -------------------
    SYSLIB    N/A         /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/copybook        N/A                     N/A
*** Physical dependencies for MortgageApplication/cobol/epsmlist.cbl:
    Library   Category        Name      Status        SourceDir/File
    --------- --------------- --------- ------------- -----------------------------------
   *SYSLIB    COPY            DFHAID    NOT RESOLVED  N/A
   *SYSLIB    COPY            EPSMLIS   NOT RESOLVED  N/A
    SYSLIB    COPY            EPSMORTF  RESOLVED      MortgageApplication/copybook/epsmortf.cpy
    SYSLIB    COPY            EPSMTINP  RESOLVED      MortgageApplication/copybook/epsmtinp.cpy
    SYSLIB    COPY            EPSMTOUT  RESOLVED      MortgageApplication/copybook/epsmtout.cpy
    SYSLIB    COPY            EPSMTCOM  RESOLVED      MortgageApplication/copybook/epsmtcom.cpy
    SYSLIB    COPY            EPSNBRPM  RESOLVED      MortgageApplication/copybook/epsnbrpm.cpy
Cobol compiler parms for MortgageApplication/cobol/epsmlist.cbl = LIB,CICS
*** Building file MortgageApplication/cobol/epscsmrt.cbl
*** Creating dependency resolver for MortgageApplication/cobol/epscsmrt.cbl with [{"library": "SYSLIB", "searchPath": [ {"sourceDir": "/u/dbehm/test-zapp/dbb-zappbuild/samples", "directory": "MortgageApplication/copybook"} ]                }] rules
*** Scanning file with the default scanner
*** Resolution rules for MortgageApplication/cobol/epscsmrt.cbl:
*** Configured resulution rules:
    Library   Category    SourceDir/File                                    Directory                           Collection              Archive
    --------- ----------- ------------------------------------------------- ----------------------------------- ----------------------- -------------------
    SYSLIB    N/A         /u/dbehm/test-zapp/dbb-zappbuild/samples          MortgageApplication/copybook        N/A                     N/A
*** Physical dependencies for MortgageApplication/cobol/epscsmrt.cbl:
    Library   Category        Name      Status        SourceDir/File
    --------- --------------- --------- ------------- -----------------------------------
    SYSLIB    COPY            EPSMTINP  RESOLVED      MortgageApplication/copybook/epsmtinp.cpy
    SYSLIB    COPY            EPSMTOUT  RESOLVED      MortgageApplication/copybook/epsmtout.cpy
    SYSLIB    COPY            EPSMTCOM  RESOLVED      MortgageApplication/copybook/epsmtcom.cpy
    SYSLIB    COPY            EPSPDATA  RESOLVED      MortgageApplication/copybook/epspdata.cpy
```

File: utilities/BuildUtilities.groovy [hunk 1]
```diff
@@ -624,3 +635,48 @@ def assertDbbBuildToolkitVersion(String currentVersion){
...
+			def libraryName = (rule.getLibrary() != null) ? rule.getLibrary().padRight(10) : "N/A".padRight(10)
+			def categoryName = (rule.getCategory() != null) ? rule.getCategory().padRight(12) : "N/A".padRight(12)
+			def srcDir = (searchPath.getSourceDir() != null) ? searchPath.getSourceDir().padRight(50) : "N/A".padRight(50)
+			def directory = (searchPath.getDirectory() != null) ? searchPath.getDirectory().padRight(36) : "N/A".padRight(36)
+			def collection = (searchPath.getCollection() != null) ? searchPath.getCollection().padRight(24) : "N/A".padRight(24)
+			def archiveFile = (searchPath.getArchive() != null) ? searchPath.getArchive().padRight(20) : "N/A".padRight(20)
+			println("    " + libraryName + categoryName + srcDir + directory + collection + archiveFile)
+
+		}
+	}
```
Comment (jbyibm) [inline utilities/BuildUtilities.groovy:664]: I can't test (no time) but can you confirm the padding?
In the output I see in the conversation the Collection and archive file are not at the correct place.
image

Comment (dennis-behm) [reply utilities/BuildUtilities.groovy:664]: I have redone the padding - See the second comment ... https://github.com/IBM/dbb-zappbuild/pull/171#issuecomment-1033974162

**Review by jbyibm — APPROVED** [review]
look good

**Pull request merged**